Merge editorialized release notes for 18.7 into `develop`

mokagio opened the pull request:
Notice the longer commit history and `RELEASE-NOTES.txt` because of:

- https://github.com/wordpress-mobile/WordPress-iOS/pull/17471
- https://github.com/wordpress-mobile/WordPress-iOS/pull/17474

I decided to wait for the editorialized release notes to make a PR against `develop` and batch all these changes because none has functional impact on the code.

mokagio committed:
Manually replace `<mark>` with `"mark"` in the release notes

App Store Connect doesn't allow markup:

> [!] An attribute value has invalid characters. - What's New In This Version in English (U.S.) must not contain markup language (for example, HTML tags). - /data/attributes/whatsNew (Spaceship::UnexpectedResponse)

mokagio committed:
Manually update en-US release notes for Jetpack

dcalhoun committed:
Merge pull request #17465 from wordpress-mobile/gutenberg/integrate_release_1.65.1

Integrate gutenberg-mobile release 1.65.1

mokagio committed:
Bump version number

mokagio committed:
Move Gutenberg 1.65.1 release note entry to 18.6.1

The fix was intended for 18.6, but I released the build (which had
already been approved on ASC) before noticing the last minute PR.

See https://github.com/wordpress-mobile/WordPress-iOS/pull/17465.

mokagio committed:
Merge 'release/18.7' into merge/release-18.6.1-into-18.7

Got conflicts on:

- `Podfile`
- `Podfile.lock`
- `WordPress/Resources/release_notes.txt`
- `config`/Version.internal.xcconfig
- `config`/Version.public.xcconfig

Resolved all of them with `git checkout --theirs` to keep the, most
recent, 18.7 version.

fluiddot committed:
Update release notes

mokagio committed:
Merge 18.6.1 hotfix  (#17471)

mokagio committed:
Update release notes with gutenberg-mobile 1.66.0 changes (#17474)

mokagio committed:
Fix 18.6.1 block position in `RELEASE-NOTES.txt`

Likely misplaced due to auto-merge conflict resolution.

mokagio committed:
Update Jetpack App Store subtitle to be less than 30 chars

mokagio committed:
Update metadata strings

mokagio committed:
Update metadata strings

mokagio commented on WordPress/Jetpack/Resources/AppStoreStrings.po:
```diff
@@ -15,7 +15,7 @@ msgstr ""
 #. translators: Subtitle to be displayed below the application name in the Apple App Store. Limit to 30 characters including spaces and commas!
 msgctxt "app_store_subtitle"
-msgid "Make your WordPress site faster and safer."
+msgid "Faster, safer WordPress"
```
As @oguzkocer raised when he covered for me releasing version 18.5, the Jetpack subtitle has been longer that 30 characters all this time.

We've discussed it internally and this is the new, shorter version at 23 characters. Hopefully all localizations will keep it below the limit 🤞

AliSoftware replied:
![image](https://user-images.githubusercontent.com/216089/142190825-4c99bd5a-e210-483f-b662-d072647aab0e.png)

mokagio commented on WordPress/Jetpack/Resources/release_notes.txt:
```diff
@@ -1,6 +1,11 @@
+- A sneak peek of the post title or parent comment, depending on whether the author left a comment on the post or replied to an existing comment
+- Rich content so you can see each comment just as the author wrote it
+- A share button for approved comments
+- Visible email and IP addresses for comment authors with Administrator or Editor roles
```
A possibly clearer version for this could be:

> - Users with Administrator or Editor roles can see the comment authors’ email and IP addresses

But I don't think it's worth chaning 🤔

AliSoftware replied:
I'd be ok with cleaning and using your version honestly.
Especially since the meaning is slightly different (and yours being closer to the actual change). The original suggests that if the comment author is an Admin/Editor then their email IP will be shown to everyone. Yours suggests that Admin/Editor will be able to see email and IP of every comment's author, which seems more correct.

AliSoftware replied:
Fixed in 22e2a30

AliSoftware commented on WordPress/Resources/AppStoreStrings.po:
```diff
@@ -34,16 +34,19 @@ msgid ""
+"- A sneak peek of the post title or parent comment, depending on whether the author left a comment on the post or replied to an existing comment\n"
+"- Rich content so you can see each comment just as the author wrote it\n"
+"- A share button for approved comments\n"
+"- Visible email and IP addresses for comment authors with Administrator or Editor roles\n"
```
Ditto about cleaning with your suggested edit above.

AliSoftware replied:
Fixed in 22e2a30

AliSoftware reviewed (approved):
✅  18.6.1 hotfix branch commits: `RELEASE-NOTES.txt` up-to-date but, as expected, no other change from the hotfix because the only change was Gutenberg `v1.65.1` but `18.7` is already on `v1.66.0`, so that checks out.
✅  `release_notes.txt` files updated for WP and JP
✅  `AppStoreStrings.po` files updated for WP and JP

LGTM, but I think it's worth applying the edit that you yourself suggested. It's a cheap change and makes sense to me by avoiding confusion or misinterpreting the impact (even if it's unlikely that we'd get, like, support tickets for people concerned about it or whatnot, better be clear anyway).
For that reason I've disabled auto-merge to let you apply it before merging.

AliSoftware reviewed (dismissed):
As discussed with @jkmassel we actually need to add some items to the notes (like the fix about MSWord), I'll push new commits in a couple of minutes to fix that and ping the relevant people, but wanted to "Request changes" ASAP in the meantime to prevent early merge

AliSoftware committed:
Update the Editorial'd Release Notes to include mention of missing Gutenberg fixes

(whose release notes missed the code-freeze deadline before being sent for Editorial)

AliSoftware commented:
@mokagio I've pushed a new commit 03daea45a8564bff7aaee40617736e170229c3f4 in order to integrate the release note items that were added in https://github.com/wordpress-mobile/WordPress-iOS/pull/17474 post-code-freeze, and were thus not part of the draft which was sent for editorial.

I've borrowed the wording for those from the Editorial'd WPAndroid notes (see https://github.com/wordpress-mobile/WordPress-Android/pull/15584) – as those were included in the draft notes and Editorial'd… before we removed the MSWord item from them.

I've also added periods at the end of each bullet point in the list.

\cc @jkmassel @twstokes @fluiddot

AliSoftware commented:
Just realized I didn't apply the suggestion about "comments email for Admin roles" item, I'll push another commit with that in a minute.
[EDIT] Done via 22e2a30. This is now ready for review again.

AliSoftware committed:
Reword the item about comment author's email+IP being visible by Admins/Editors

AliSoftware commented on WordPress/Jetpack/Resources/release_notes.txt:
```diff
@@ -1,6 +1,13 @@
-- We fixed more cases where embed blocks would show "unable to display content", allowing you to see all of your content.
+Two-step authentication now comes with extra security. You'll need to unlock your device to approve or deny notifications.
+We added Media File and Attachment Page options to the image block's Link To menu so you can set commonly used URLs.
+We also fixed an issue that made the interface crash when users copied and pasted lists from Microsoft Word.
```
I considered moving this added line at the very end, it being a bugfix as opposed to the other items being feature improvements. IMHO it would have put less emphasis on the bugfix if I did so, which could have been nice.

But otoh, the wording that was originally provided by Editorial ends with "Finally, we updated the UI […]" so didn't seem right to add an item / line after that "Finally" line…

Any suggestions to find a better balance welcome.

jkmassel replied:
```suggestion
We fixed an issue that made the interface crash when users copied and pasted lists from Microsoft Word.
```

jkmassel replied:
```suggestion
We fixed an issue that made the interface crash when users copied and pasted lists from Microsoft Word.
```

To align with what's in the `.po` file?

AliSoftware replied:
Nice catch! I indeed messed up between various copy/pastes; removed the `also` in b305045.

fluiddot commented on RELEASE-NOTES.txt:
```diff
@@ -1,12 +1,17 @@
 * [***] Two-step Authentication notifications now require an unlocked device to approve or deny them.
 * [***] Site Comments: Updated comment details with a fresh new look and capability to display rich contents. [#17466]
+* [**] Block editor: Image block: Add ability to quickly link images to Media Files and Attachment Pages [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3971]
+* [**] Block editor: Fixed a crash that could occur when copying lists from Microsoft Word. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/4174]
```
This entry is duplicated, as this issue was fixed in `18.6.1`, so we can remove it.

AliSoftware replied:
Yeah I'm not sure how @mokagio usually handles this.

In WPAndroid I've never added a `RELEASE-NOTES.txt` entry for hotfixes like he did for `18.6.1` (because those release notes don't go thru the usual process for hotfixes anyway given their timely nature), so we don't have the issue as we don't have entries for `18.6.1` anyway.

In WPiOS Gio started to do this for hotfixes which has its justifications: I actually like it given that `RELEASE-NOTES.txt` is also for us devs to keep track of when each feature/fix was shipped). But then this raises the question as to how we should cherry-pick the release notes for those cases, as it adds extra churn (both release notes will always be present in that case, once coming from the `release/18.6.1` hotfix branch since we started to add sections for those in WPiOS, and another copy coming from the `develop` branch that the `release/18.7` is cut from, before the hotfix branch gets merged later…).

So that means that when we merge back the `release/18.6.1` hotfix branch into `release/18.7` current release to resync everything, we'd have to add an extra commit to clean up the `RELEASE-NOTES.txt` file as well to remove the duplicates… not ideal either.

---

I'm ok either way (to clean them up and remove those 2 entries from that `REALEASE-NOTES.txt` file here as you suggest, or to keep them to enforce the idea that they were cherry-picked in both hotfix and final?)… but I'll let @mokagio decide on this one given how he handles hotfix release notes in general.

jkmassel replied:
I'm ok to mention it with this release so folks know it's been fixed (because the note wouldn't be translated for folks in 18.6.1, so they may not know).

Thanks @fluiddot!!

AliSoftware replied:
Oh we're not talking about removing those from the 18.7 release notes in `AppStoreStrings.po` itself, we all agree the entries should still go into the 18.7 release notes that will be translated and displayed to end users on AppStore 🙂

The question was more about should we remove the duplicate entry in the developer-facing `RELEASE-NOTES.txt` (even if we keep it in the `.po` files) 🤷 

That's not critical either way given that's only the internal notes file (which is why I was inclined to let @mokagio do as he prefers)

mokagio replied:
I've decided to remove the "duplicated" entry. More on the rationale here: [`b2ae4fc` (#17483)](https://github.com/wordpress-mobile/WordPress-iOS/pull/17483/commits/b2ae4fcfce7f0d81c82a7dc06f9f2f80ac7474f8)

If this was a library, I would have pushed back, but I think in this case simplicity trumps my own pedantry.

AliSoftware committed:
Align wording in `release_notes.txt` with `.po`

jkmassel reviewed (approved):
I'm good with this, but I'll let @mokagio merge

mokagio commented:
<img width="839" alt="Screen Shot 2021-11-18 at 2 03 01 pm" src="https://user-images.githubusercontent.com/1218433/142344207-adb15eb6-8334-40f8-afc3-0c9b2a34e0a4.png">

😭

mokagio commented:
Thanks @AliSoftware @jkmassel @fluiddot for the help moving this forward.

Going to solve the merge conflicts, give the diff a last pass, and merge. Unfortunately, this will need to be done in a _new_ PR.

mokagio committed:
Remove dedicated release note entry for 18.6.1

See conversation here:
https://github.com/wordpress-mobile/WordPress-iOS/pull/17483#discussion_r751494058

My rationale for adding it was to track the reason for 18.6.1 in
`RELEASE-NOTES.txt`.

My choice raised a few questions and doesn't align with what other apps
do. For the sake of simplicity, it might be best to revert.

We have internal documentation for each release, GitHub PRs, and the
possibility to use Git to diff between tags (e.g. `git diff 18.6
18.6.1`). That information won't be lost and only a few extra keystrokes
away.

mokagio committed:
Tweak the Media File / Link To release note after translator's feedback

See https://wordpress.slack.com/archives/C02RQC4LY/p1637173139014900

mokagio committed:
Update metadata strings

mokagio committed:
Update metadata strings

AliSoftware commented:
Closing this one in favor of https://github.com/wordpress-mobile/WordPress-iOS/pull/17492 which fixes the conflict
